include/grpc/impl/codegen/grpc_types.h

```diff
@@ -332,7 +333,12 @@ typedef struct grpc_op {
     struct {
       size_t count;
       grpc_metadata *metadata;
-      grpc_compression_level compression_level;
+      /** If \a is_set, \a compression_level will be used for the call.
+       * Otherwise, \a compression_level won't be considered */
+      struct {
+        bool is_set;
+        grpc_compression_level compression_level;
+      } maybe_compression_level;
```

Comment:
@ctiller do you think it'd be worth it to encode the `is_set` bit within `compression_level` itself with some bit twiddling as we already do in other places? I don't think the small size increase of the op is important, given how it's translated into a stream_op anyway...